### PR TITLE
Port failure classes from Java SDK

### LINKED ISSUE #146: [Feature Request] Port Workflow error classes from Java SDK
- For instance for activity failures.
- Errors should have an `isRetryable` attributes.

### PR DESCRIPTION
**Pull request by bergundy**
Closes #146

We still need to figure out what to do for activity cancellation.

### CODE REVIEW

**Comment by bergundy** [comment]
Depends on https://github.com/temporalio/sdk-core/pull/159

File: packages/common/src/failure.ts [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+export const RetryState = coresdk.common.RetryState;
+
+/**
+ * Represents failures that can cross Workflow and Activity boundaries.
+ *
+ * Only exceptions that extend this class will be propagated to the caller.
+ *
+ * **Never extend this class or any of its derivatives.** They are to be used by the SDK code
+ * only. Throw an instance {@link ApplicationFailure} to pass application specific errors between
+ * Workflows and Activities.
```
Comment (Sushisource) [inline packages/common/src/failure.ts:17]: Could just make this constructor private and then expose a `create` function to allow constructing them for testing purposes. That way the class can't be extended.

**Review by Sushisource — APPROVED** [review]

**Comment by bergundy** [comment]
Note that this addition of 3 proto files increased the generated JS file from 244kb to 638kb, at some point we might want to optimize this but I'll let this go because looks like perf is still performant enough from a short test I made.

**Comment by sw-yx** [comment]
good that you are watching the perf though, nice

**Pull request merged**